### PR TITLE
[Merged by Bors] - feat: port Algebra.Order.Monoid.[TypeTags, Units]

### PR DESCRIPTION
**Pull request by zeramorphic**
mathlib3 SHA: f1a2caaf51ef593799107fe9a8d5e411599f3996

### CODE REVIEW

File: Mathlib/Algebra/Order/Monoid/TypeTags.lean [hunk 1]
```diff
@@ -0,0 +1,163 @@
+/-
+Copyright (c) 2016 Jeremy Avigad. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jeremy Avigad, Leonardo de Moura, Mario Carneiro, Johannes Hölzl
+-/
+import Mathlib.Algebra.Group.TypeTags
+import Mathlib.Algebra.Order.Monoid.Cancel.Defs
+import Mathlib.Algebra.Order.Monoid.Canonical.Defs
+
+/-! # Ordered monoid structures on `multiplicative α` and `additive α`. -/
```
Comment (Ruben-VandeVelde) [inline Mathlib/Algebra/Order/Monoid/TypeTags.lean:10]:
```suggestion
/-! # Ordered monoid structures on `Multiplicative α` and `Additive α`. -/
```

File: Mathlib/Algebra/Order/Monoid/Units.lean [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  PartialOrder.lift val Units.ext
+
+@[to_additive]
+instance [Monoid α] [LinearOrder α] : LinearOrder αˣ :=
+  LinearOrder.lift' val Units.ext
+
+/-- `val : αˣ → α` as an order embedding. -/
+@[to_additive "`val : add_units α → α` as an order embedding.",
+  simps (config := { fullyApplied := false })]
+def orderEmbedding_val [Monoid α] [LinearOrder α] : αˣ ↪o α :=
```
Comment (Ruben-VandeVelde) [inline Mathlib/Algebra/Order/Monoid/Units.lean:42]: Should this be `orderEmbeddingVal`? I don't think there should be underscores in a `def`

**Review by Ruben-VandeVelde — APPROVED** [review]
lgtm, two comments

**Comment by zeramorphic** [comment]
Thanks for the comments - I misread the new naming convention guide.

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**